### PR TITLE
LibWeb: Implement the `<use>` and `<symbol>` SVG elements

### PR DESCRIPTION
**Pull request by PrestonLTaylor**
This pr implements the `<use>` and `<symbol>` SVG elements!

This is my first kind-of major change so feedback would be appreciated! :^)

Spec Links: [use](https://svgwg.org/svg2-draft/struct.html#UseElement) [symbol](https://svgwg.org/svg2-draft/struct.html#SymbolElement)

Example:
```html
<svg viewBox="0 0 30 10" xmlns="http://www.w3.org/2000/svg">
    <symbol id="myCircle">
        <circle cx="5" cy="5" r="4" stroke="blue" />
    </symbol>
    <use href="#myCircle" fill="blue" />      
    <use href="#myCircle" x="20" fill="white" stroke="red" />
</svg>
```
Image of feature in ladybird

### CODE REVIEW

File: Userland/Libraries/LibWeb/SVG/SVGUseElement.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    auto href = attribute(SVG::AttributeNames::href);
+
+    // FIXME: Support loading of external svg documents
+    auto referenced_element = document().query_selector(href.view()).release_value_but_fixme_should_propagate_errors();
+    if (!referenced_element || !is_valid_reference_element(referenced_element)) {
+        return;
+    }
+
+    // The ‘use’ element references another element, a copy of which is rendered in place of the ‘use’ in the document.
+    auto cloned_reference_node = referenced_element->clone_node(nullptr, true);
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/SVG/SVGUseElement.cpp:83]: If the referenced element subtree is mutated with JavaScript, aren't those mutations supposed to be visible in all the `<use>` instances?

Comment (PrestonLTaylor) [reply Userland/Libraries/LibWeb/SVG/SVGUseElement.cpp]: Implemented as discussed on discord.

`SVGElement::inserted()` etc. now call into any `SVGUseElement`s and the use element checks if it should update its shadow-tree.

This now allows for the mutations in JavaScript to be visible!

Example Code:
```html
  <svg viewBox="0 0 30 10" xmlns="http://www.w3.org/2000/svg">
    <symbol id="myCircle">
    </symbol>
    <use href="#myCircle" fill="blue" />
    <use href="#myCircle" x="20" fill="white" stroke="red" />
  </svg>
  <svg id="delete"></svg>
  <script>
          let symbol = document.querySelector("symbol");
          function insert() {
                  let circle = document.createElementNS("http://www.w3.org/2000/svg", "circle");
                  circle.setAttribute("cx", "5");
                  circle.setAttribute("cy", "5");
                  circle.setAttribute("r", "4");
                  circle.setAttribute("stroke", "blue");
                  symbol.appendChild(circle);
          }
  
          function change_attribute() {
                  let circle = symbol.querySelector("circle");
                  circle.setAttribute("stroke", "red");
          }
  
          function remove() {
                  symbol.remove();
          }
  
          function remove_unrelated_svg() {
                  let svg = document.querySelector("#delete");
                  svg.remove();
          }
  
          setTimeout(insert, 2000);
          setTimeout(change_attribute, 4000);
          setTimeout(remove_unrelated_svg, 6000);
          setTimeout(remove, 8000);
  </script>
```

File: Userland/Libraries/LibWeb/SVG/SVGUseElement.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+public:
+    virtual ~SVGUseElement() override = default;
+
+    virtual void parse_attribute(DeprecatedFlyString const& name, DeprecatedString const& value) override;
+
+    virtual void inserted() override;
+
+    void svg_element_changed(SVGElement*);
+    void svg_element_removed(SVGElement*);
```
[the remark this answers is not in the document]
Comment (PrestonLTaylor) [reply Userland/Libraries/LibWeb/SVG/SVGUseElement.h]: Good catch!

File: Userland/Libraries/LibWeb/SVG/SVGElement.cpp [hunk 3]
```diff
@@ -34,4 +37,56 @@ void SVGElement::visit_edges(Cell::Visitor& visitor)
...
+    remove_from_use_element_that_reference_this();
+}
+
+void SVGElement::remove_from_use_element_that_reference_this()
+{
+    if (is<SVGUseElement>(this)) {
+        return;
+    }
+
+    document().for_each_in_subtree_of_type<SVGUseElement>([this](SVGUseElement& use_element) {
```
[the remark this answers is not in the document]
Comment (PrestonLTaylor) [reply Userland/Libraries/LibWeb/SVG/SVGElement.cpp:86]: The `<use>` element is allowed to reference elements from other svg subtrees.

For example you can have something like this:

```html
<svg>
    <symbol id="test"><circle cx="5" cy="5" r="4" stoke="blue"></symbol>
    <use href="#test" />
</svg>
<svg>
    <use href="#test" />
</svg>
<svg>
    <use href="#test" />
</svg>
<script>
function change() {
    let circle = document.querySelector("circle");
    circle.setAttribute("stroke", "red");
}
setTimeout(change, 1000);
</script>
```

All of the `<use>` elements should be updated to have a stroke of red.

**Pull request merged**